tests/ci/common_posix_setup.sh

```diff
@@ -143,63 +143,31 @@ function build_and_run_minimal_test {
   run_cmake_custom_target 'run_minimal_tests'
 }
 
-function print_system_and_dependency_information {
-  if command -v cmake &> /dev/null
-  then
-    echo ""
-    echo "CMake version:"
-    cmake --version
-  fi
-  if command -v cmake3 &> /dev/null
-  then
-    echo ""
-    echo "CMake version (cmake3 executable):"
-    cmake3 --version
-  fi
-  if command -v go &> /dev/null
-  then
-    echo ""
-    echo "Go version:"
-    go version
-  fi
-  if command -v perl &> /dev/null
-  then
-    echo ""
-    echo "Perl version:"
-    perl --version
-  fi
-  # Ninja executable names are not uniform
-  if command -v ninja-build &> /dev/null
-  then
-    echo ""
-    echo "Ninja version"
-    ninja-build --version
-  elif command -v ninja &> /dev/null
-  then
-    echo ""
-    echo "Ninja version"
-    ninja --version
-  fi
-  if command -v gcc &> /dev/null
-  then
-    echo ""
-    echo "GCC version:"
-    gcc --version
-    echo ""
-    echo "G++ version:"
-    g++ --version
-  elif command -v clang &> /dev/null
-  then
-    echo ""
-    echo "Clang version:"
-    clang --version
-  fi
-  if command -v make &> /dev/null
+function print_executable_information {
+  EXE_NAME=${1}
+  EXE_ARGUMENT=${2}
+  LABEL=${3}
+
+  if command -v ${EXE_NAME} &> /dev/null
   then
     echo ""
-    echo "Make version:"
-    make --version
+    echo "${LABEL}:"
+    ${EXE_NAME} ${EXE_ARGUMENT}
   fi
+}
+
+function print_system_and_dependency_information {
+  print_executable_information "cmake" "--version" "CMake version"
+  print_executable_information "cmake3" "--version" "CMake version (cmake3 executable)"
+  print_executable_information "go" "version" "Go version"
+  print_executable_information "perl" "--version" "Perl version"
+  # Ninja executable names are not uniform over operating systems
+  print_executable_information "ninja-build" "--version" "Ninja version (ninja-build executable)"
+  print_executable_information "ninja" "--version" "Ninja version (ninja executable)"
+  print_executable_information "gcc" "--version" "GCC version"
+  print_executable_information "g++" "--version" "G++ version"
+  print_executable_information "clang" "--version" "Clang version"
```

Comment:
You specifically look for `gcc`, `g++` and `clang`, but what about verifying the version of `clang++`? Should we check the whether/which of system defaults: `cc` and `c++`?

Reply:
Maybe we can skip `gcc/++` or `clang/++` because CMake build logs tell which compilers are used
```
+ cmake -DBUILD_SHARED_LIBS=1 -DCMAKE_BUILD_TYPE=Release -GNinja ../
-- The C compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/gcc-7
-- Check for working C compiler: /usr/bin/gcc-7 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/g++-7
-- Check for working CXX compiler: /usr/bin/g++-7 -- works
```

Reply:
I think we should just add all the stuff we can think about to see what environment we are running in.